docs/application/native/guides/text-input/capi-ui-gesture.md

```diff
@@ -0,0 +1,248 @@
+        if (HAND_GESTURE_ERROR_NONE != ret)
+            /* Error handling */
+    }
+    ```
+
+<a name="set_option"></a>
+## Setting an Option
+
+You can select an option to detect hand gestures continuously or not. If you would like to make the application detect hand gestures continuously, set the option as `HAND_GESTURE_OPTION_ALWAYS_ON` using `hand_gesture_set_option()` function.
+
+    ```
+    void
+    set_option(hand_gesture_h gesture_h)
+    {
+        int ret;
+        ret = hand_gesture_set_option(gesture_h, HAND_GESTURE_OPTION_ALWAYS_ON);
+        if (HAND_GESTURE_ERROR_NONE != ret)
+            /* Error handling */
+    }
+    ```
+
+
+<a name="callback"></a>
+## Setting and Unsetting Callbacks
+
+The Gesture provides two types of callbacks for getting error and gesture recognition result.
+The enum values, as well as the parameter details, for the callback parameters are defined in the `gesture.h` and `gesture_common.h` header files.
+
+1. Error
+
+   When an error occurs, the Gesture library sends a message using a callback:
+
+   > **Note**
+   >
+   > Please set the error callback function before calling `hand_gesture_start_recognition()`.
+
+    ```
+    /* Callback */
+    void
+    error_cb(hand_gesture_h gesture_h, hand_gesture_error_e reason, const char* msg, void* user_data)
+    {
+        /* Your code */
+    }
+
+    /* Set */
+    void
+    set_error_cb(hand_gesture_h gesture_h)
+    {
+        int ret;
+        ret = hand_gesture_set_error_cb(gesture_h, error_cb, NULL);
+        if (HAND_GESTURE_ERROR_NONE != ret)
+            /* Error handling */
+    }
+
+    /* Unset */
+    void
+    unset_error_cb(hand_gesture_h gesture_h)
+    {
+        int ret;
+        ret = hand_gesture_unset_error_cb(gesture_h);
+        if (HAND_GESTURE_ERROR_NONE != ret)
+            /* Error handling */
+    }
+    ```
+
+2. Recognition result
+
+   When the hand gesture is recognized, the recognized result is delivered through the `hand_gesture_recognition_cb()`.
+
+   You can [set the recognition result callback function using `hand_gesture_start_recognition()`](#start_stop).
+
+
+<a name="start_stop"></a>
+## Start and Stop recognizing the gesture
+
+1. To start recognizing the gesture, use the `hand_gesture_start_recognition()` function.
+
+   When you start the gesture recognition, you should input the gesture type targeted to be recognized. Before calling `hand_gesture_start_recognition()`, please check whether the target gesture type is supported or not with `hand_gesture_is_supported_type()`.
+
+   After calling `hand_gesture_start_recognition()`, the recognition results will be coming through the callback function `hand_gesture_recognition_cb()`.
+   When the callback function is invoked, you can handle the recognition results by corresponding to some operations.
+   For example, when `HAND_GESTURE_WRIST_UP` is occurred, you can turn on the screen of the wearable device.
```

Comment:
Please remove the period and use a colon.
  
For example, when `HAND_GESTURE_WRIST_UP` is occurred, you can turn on the screen of the wearable device:

Reply:
Done.